lib/pbench/server/api/resources/query_apis/search_api.py

```diff
@@ -0,0 +1,224 @@
+                    username within a given time range
+
+            "fields": List of run document fields end user is interested in. E.g.
+                    ["@timestamp", "run.controller", "run.name", ...]
+                    The fields controls what fields are returned by the query, and it has no effect
+                    on which fields are searched.
+        """
+        user = json_data.get("user")
+        start = json_data.get("start")
+        end = json_data.get("end")
+
+        # If the search_term parameter is not specified we will return all the Elasticsearch hits
+        search_term = json_data.get("search_term", "")
+        # If no fields are specified, the query will return all the fields from Elasticsearch hits
+        selected_fields = json_data.get("fields", [])
+
+        # We need to pass string dates as part of the Elasticsearch query; we
+        # use the unconverted strings passed by the caller rather than the
+        # adjusted and normalized datetime objects for this.
+        start_arg = f"{start:%Y-%m}"
+        end_arg = f"{end:%Y-%m}"
+
+        self.logger.info(
+            "Search query for user {}, prefix {}: ({} - {}) on query: {}",
+            user,
+            self.prefix,
+            start,
+            end,
+            search_term,
+        )
+
+        uri_fragment = self._gen_month_range("run", start, end)
+        self.logger.info("fragment, {}", uri_fragment)
+        return {
+            "path": f"/{uri_fragment}/_search",
+            "kwargs": {
+                "json": {
+                    "query": {
+                        "bool": {
+                            "filter": [
+                                {"term": self._get_user_term(json_data)},
+                                {
+                                    "range": {
+                                        "@timestamp": {"gte": start_arg, "lte": end_arg}
+                                    }
+                                },
+                                {"query_string": {"query": f"*{search_term}*"}},
+                            ]
+                        }
+                    },
+                    "sort": [{"@timestamp": {"order": "desc"}}],
+                    "_source": {"include": selected_fields},
+                },
+                "params": {"ignore_unavailable": "true"},
+            },
+        }
+
+    def postprocess(
+        self, es_json: Dict[AnyStr, Any], context: CONTEXT
+    ) -> Dict[AnyStr, Any]:
+        """
+        Returns a summary of the returned Elasticsearch query results, showing
+        the list of dictionaries with user selected fields from request json as keys
+
+        [
+            {
+                "controller": "alphaville.example.com",
+                "results": 2,
+                "last_modified_value": 1598473155810.0,
+                "last_modified_string": "2020-08-26T20:19:15.810Z"
+            }
+        ]
```

Comment:
This looks like the example from `controllers/list`, not `index/search`! (Except, interestingly, with `"key"` removed, which we *do* want to remove from `controllers/list` and make the dashboard effect add it ... but not here. 😁 )